server/src/main/java/io/unitycatalog/server/service/IcebergRestCatalogService.java

```diff
@@ -176,18 +137,17 @@ public HttpResponse tableExists(
     }
   }
 
-  @Get("/v1/namespaces/{namespace}/tables/{table}")
+  @Get("/v1/catalogs/{catalog}/namespaces/{namespace}/tables/{table}")
```

Comment:
What is the purpose of a`catalog` part here? The Iceberg specification doesn't have such concept, and this change would further deviate from the spec defined at https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L240-L254
Parent namespaces should be nested into a `parent=` GET parameter to conform the specification.

Reply:
This does not deviate from the spec and is valid use of {prefix} as defined in the spec.

Reply:
> Well isn't catalog part of the namespace here
We are/would be saying that catalog is NOT considered part of the namespace.

> I don't think {prefix} was really meant to be used this way.
Again, this valid to the spec where operations are in the context of a catalog.

Reply:
> catalog is NOT considered part of the namespace.

Then it makes sense of course.
But: catalog did use to be part of the namespace and on top of that, a client would be restricted to work with only one catalog from now on. That strikes me as a rather larger structural change compared to just a namespace separator fix, wouldn't you agree?

Also - how is this going to be backward compatible with older client versions? (Considering those that provided `.` as separator of course..)

Reply:
> But: catalog did use to be part of the namespace and on top of that, a client would be restricted to work with only one catalog from now on. That strikes me as a rather larger structural change compared to just a namespace separator fix, wouldn't you agree?

Not if we felt the original implementation was incorrect. Other IRC implementations (Tabular, Polaris) utilize the prefix in this way and separate out a entity like catalog/warehouse and do not consider it to be part of the namespace so this is not a new/foreign concept.

I agree that using the . was not to spec for namespace separating, but since we consider the catalog to not be part of the namespace and UC OSS only has 1 level for namespace (Schema), we don't have to worry about a separator.

> Also - how is this going to be backward compatible with older client versions? (Considering those that provided . as separator of course..)

0.2 hasn't even been released yet, so not sure why anyone would expect it to be "stable" as far as things like this. 
Also, per the spec, REST clients should be hitting the config endpoint first to get configurations.

> one more thing: I belive this also takes away the possibility for clients to discover what catalogs there are? Should we plan to substitute the lost functionality somehow?

UC OSS has other endpoints for this that are not part of the IRC spec. (so this can be done through the cli, UI, etc.)

Reply:
Okay, thanks for explaining @alexreid-db